qulice-checkstyle/src/test/java/com/qulice/checkstyle/RedundantThrowsCheckTest.java

```diff
@@ -0,0 +1,152 @@
+     */
+    private static final String LICENSE_PROP = "license";
+
+    /**
+     * License rule.
+     *
+     * @checkstyle VisibilityModifierCheck (5 lines)
+     */
+    @Rule
+    public final transient LicenseRule rule = new LicenseRule();
+
+    /**
+     * Tries to violates RedundantThrowsCheck with a test class with
+     * inner Exception descendant.
+     *
+     * @throws Exception If something wrong happens inside
+     */
+    @Test
+    public void violatesRedundantThrowsCheckWithInnerException()
+        throws Exception {
+        final Environment.Mock mock = new Environment.Mock();
+        final File license = this.rule
+            .savePackageInfo(new File(mock.basedir(), "src/main/java/foo"))
+            .withLines(new String[]{"HELLO"})
+            .withEol("\n")
+            .file();
+        final String valid = Joiner.on("\n").join(
+            "/**",
+            " * HELLO",
+            " */",
+            "package foo;",
+            "/**",
+            " * A.",
+            " * @author Eugene Bukhtin (maurezen@gmail.com)",
+            " * @version $Id$",
+            " */",
+            "public class A {",
+            "    /**",
+            "     * Test illustration method.",
+            "     * @throws E no problem here",
+            "     */",
+            "    public final void mthd() throws E {",
+            "        throw new E(\"just a demo\");",
+            "    }",
+            "",
+            "    /**",
+            "     * E.",
+            "     */",
+            "    public static class E extends Exception {",
+            "        /**",
+            "         * Ctor.",
+            "         * @param message Message",
+            "         */",
+            "        public E(final String message) {",
+            "            super(message);",
+            "        }",
+            "    }",
+            "}",
+            ""
+        );
+        final String invalid = Joiner.on("\n").join(
+            "/**",
+            " * HELLO",
+            " */",
+            "package foo;",
+            "/**",
+            " * B.",
+            " * @author Eugene Bukhtin (maurezen@gmail.com)",
+            " * @version $Id$",
+            " */",
+            "public class B extends A {",
+            "    /**",
+            "     * Test illustration method.",
+            "     * @throws E THIS CAUSES THE PROBLEM",
+            "     */",
+            "    public final void method() throws E {",
+            "        throw new E(\"just a demo\");",
+            "    }",
+            "}",
+            ""
+        );
+        final Environment env = mock
+            .withParam(
+                RedundantThrowsCheckTest.LICENSE_PROP,
+                this.toURL(license)
+        )
+            .withFile("src/main/java/foo/A.java", valid)
+            .withFile("src/main/java/foo/B.java", invalid);
+        new CheckstyleValidator().validate(env);
```

Comment:
This check should throw exception for invalid and should NOT throw it for valid.  You are not checking that.  I think it will be better to separate those cases.